tests_adapters/test_adapter_composition.py

```diff
@@ -245,6 +246,33 @@ def test_batch_split_with_heads(self):
             )
         )
 
+    def test_parallel_generate(self):
+        if self.config_class not in ADAPTER_MODEL_MAPPING or (
+            not hasattr(ADAPTER_MODEL_MAPPING[self.config_class], "add_seq2seq_lm_head")
+            and not hasattr(ADAPTER_MODEL_MAPPING[self.config_class], "add_causal_lm_head")
+        ):
+            self.skipTest("No seq2seq or causal language model head")
+
+        model1 = AutoAdapterModel.from_config(self.config())
+        model1.add_adapter("adapter1")
+        model1.add_adapter("adapter2")
+        if hasattr(model1, "add_seq2seq_lm_head"):
+            model1.add_seq2seq_lm_head("adapter1")
+            model1.add_seq2seq_lm_head("adapter2")
+        else:
+            model1.add_causal_lm_head("adapter1")
+            model1.add_causal_lm_head("adapter2")
```

Comment:
If a model has both a seq2seq_lm_head and a causal_lm_head, the test only checks if `model.generate()` works with the seq2seq_lm_head, but not if it works with the causal_lm_head.
However, I'm not sure if it's necessary to test both heads.

Reply:
I think we currently don't have a model that can have both heads. Either the model is encoder-decoder, then it has a seq2seq generation head, or it is decoder-only, then it has a causal generation head.

Reply:
Ok, then it is most likely not necessary to test both cases.